ic-agent/src/agent/mod.rs

```diff
@@ -1275,3 +1278,31 @@ impl<'agent> UpdateBuilder<'agent> {
         })
     }
 }
+
+#[cfg(all(test, feature = "reqwest"))]
+mod offline_tests {
+    use super::*;
+    // Any tests that involve the network should go in agent_test, not here.
+
+    #[test]
+    fn rounded_expiry() {
+        let agent = Agent::builder()
+            .with_url("http://not-a-real-url")
+            .build()
+            .unwrap();
+        let mut prev_expiry = None;
+        let mut num_timestamps = 0;
+        for _ in 0..6 {
```

Comment:
Just out of curiosity, why aren't we using the iterator from the for loop instead of num_timestamps?